### PR TITLE
feat: add 0x34-CALLVALUE opcode

### PR DESCRIPTION
**Pull request by ftupas**
## Pull Request type

Please check the type of change your PR introduces:

- [ ] Bugfix
- [x] Feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no API changes)
- [ ] Build-related changes
- [ ] Documentation content changes
- [ ] Other (please describe):

## What is the current behavior?

Resolves: #120 

## What is the new behavior?

- 0x34 CALLVALUE opcode
- test for some value and stack overflow

## Does this introduce a breaking change?

- [ ] Yes
- [X] No

## Added changes to [`CHANGELOG.md`](../CHANGELOG.md)

- [X] Yes
- [ ] No

### CODE REVIEW

File: crates/evm/src/tests/test_utils.cairo [hunk 1]
```diff
@@ -16,10 +20,14 @@ fn zero_address() -> ContractAddress {
     0.try_into().unwrap()
 }
 
+fn callvalue() -> u256 {
+    123456789
```
Comment (ftupas) [inline crates/evm/src/tests/test_utils.cairo:24]: Value reflected in playground

File: crates/evm/src/tests/test_utils.cairo [hunk 2]
```diff
@@ -38,6 +46,21 @@ fn setup_execution_context() -> ExecutionContext {
     )
 }
 
+fn setup_execution_context_max_stack_depth() -> ExecutionContext {
```
Comment (ftupas) [inline crates/evm/src/tests/test_utils.cairo:49]: Added this helper for initializing a context with max stack depth for opcodes to test for stack overflow

File: crates/evm/src/tests/test_utils.cairo (hunk 1 above)
Comment (Eikix) [reply crates/evm/src/tests/test_utils.cairo:24]: what do you mean

File: crates/evm/src/tests/test_utils.cairo (hunk 2 above)
Comment (ftupas) [reply crates/evm/src/tests/test_utils.cairo:49]: Yeah indeed we have this overflow test already [here](https://github.com/kkrt-labs/kakarot-ssj/blob/56daa7cbf6cf75f39789d46f2e4d55e4fe4f6ea5/crates/evm/src/tests/test_stack.cairo#L80), I guess I was trying to make sure that this error is being propagated but I agree I think it should suffice 🤙 let me remove this

File: crates/evm/src/tests/test_utils.cairo (hunk 1 above)
Comment (ftupas) [reply crates/evm/src/tests/test_utils.cairo:20]: Sorry I meant [evm.codes](https://www.evm.codes/#34?fork=shanghai)

Example

<h2 dir="auto"></h2>

  | Input | Output
-- | -- | --
1 |   | 123456789

**Comment by Eikix** [comment]
You need to delete changelog

**Review by Eikix — APPROVED** [review]
lgtm ✌️🦾

**Pull request merged**